synapse/handlers/presence.py

```diff
@@ -1430,37 +1430,86 @@ async def _handle_state_delta(self, room_id: str, deltas: List[JsonDict]) -> Non
         #   2. presence states of newly joined users to all remote servers in
         #      the room.
         #
-        # TODO: Only send presence states to remote hosts that don't already
+        # However, only send presence states to remote hosts that don't already
         # have them (because they already share rooms).
 
         # Get all the users who were already in the room, by fetching the
         # current users in the room and removing the newly joined users.
         users = await self.store.get_users_in_room(room_id)
         prev_users = set(users) - newly_joined_users
 
+        async def is_only_room_shared(current_user_id: str, other_user_id: str) -> bool:
```

Comment:
The name of this function is a little confusing, `is_only_one_room_shared` or `users_have_one_mutual_room` or something else might be clearer.

Reply:
I almost went with `is_this_only_room_shared` but I try and keep 'this' out of function names after being exposed to so many languages that use 'this' as a keyword. Since the context of the function is comparing the room being checked joined, I thought it would be ok. I like your first suggestion better.

Reply:
Sorry that took a minute
Fixed in [6e090b1](https://github.com/matrix-org/synapse/pull/15818/commits/6e090b18ddde7fb443c97208d64c438936846253)